src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -250,6 +262,83 @@ private static IntPtr GetSslMethod(bool isServer, long options)
+                    chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+
+                    using (SafeX509StackHandle chainStack = libcrypto.X509_STORE_CTX_get1_chain(storeHandle))
+                    {
+                        if (chainStack.IsInvalid)
+                        {
+                            Debug.Fail("Invalid chain stack handle");
+                            return 0;
+                        }
+
+                        IntPtr certPtr = Crypto.GetX509StackField(chainStack, 0);
+                        if (IntPtr.Zero == certPtr)
+                        {
+                            return 0;
+                        }
+
+                        using (X509Certificate2 cert = new X509Certificate2(certPtr))
+                        {
+                            return chain.Build(cert) ? 1 : 0;
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void UpdateCAListFromRootStore(libssl.SafeSslContextHandle context)
+        {
+            using (SafeX509NameStackHandle nameStack = Crypto.NewX509NameStack())
+            {
+                //Enumerate Certificates from LocalMachine and CurrentUser root store 
+                AddX509Name(nameStack, StoreLocation.LocalMachine);
+                AddX509Name(nameStack, StoreLocation.CurrentUser);
+                libssl.SSL_CTX_set_client_CA_list(context, nameStack);
+
+                // The handle ownership has been transferred into the CTX.
+                nameStack.SetHandleAsInvalid();
+            }
+
+        }
+
+        private static void AddX509Name(SafeX509NameStackHandle nameStack, StoreLocation storeLocation)
+        {
+            using (var myStore = new X509Store(StoreName.Root, storeLocation))
+            {
+                myStore.Open(OpenFlags.ReadOnly);
+
+                foreach (var cert in myStore.Certificates)
+                {
+                    using (SafeX509Handle certHandle = new SafeX509Handle(cert.Handle, false))
+                    {
+                        using (SafeX509NameHandle nameHandle = libcrypto.X509_NAME_dup(libcrypto.X509_get_issuer_name(certHandle)))
+                        {
+                            //Validate if X509_NAME is not present in the stack
+                            //Avoiding duplicate Names
+                            if (Crypto.FindX509NameInStack(nameStack, nameHandle) < 0)
+                            {
+                                if (!Crypto.PushX509NameStackField(nameStack, nameHandle))
+                                {
+                                    throw new CryptographicException(SR.net_ssl_x509Name_push_failed_error);
+                                }
+                            }
+                         
+                            // The handle ownership has been transferred into the STACK_OF(X509_NAME).
+                            nameHandle.SetHandleAsInvalid();
```

Comment:
This should be inside the if-not-duplicate check, effectively as the else to PushX509NameStackField failing.

In fact, it might be better written as

```C#
If (Crypto.PushX509NameStackField(nameStack, nameHandle))
{
    // The handle ownership has been transferred into the STACK_OF(X509_NAME).
    nameHandle.SetHandleAsInvalid();
}
else
{
    throw new CryptographicException(SR.net_ssl_x509Name_push_failed_error);
}